generative/maisi/maisi_inference_tutorial.ipynb

```diff
@@ -0,0 +1,288 @@
+    "\n",
+    "mask_generation_diffusion_unet = define_instance(args, \"mask_generation_diffusion_def\").to(device)\n",
+    "checkpoint_mask_generation_diffusion_unet = torch.load(args.trained_mask_generation_diffusion_path)\n",
+    "mask_generation_diffusion_unet.load_state_dict(checkpoint_mask_generation_diffusion_unet, strict=True)\n",
+    "mask_generation_scale_factor = args.mask_generation_scale_factor\n",
+    "\n",
+    "print(\"All the trained model weights have been loaded.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9125f7c8",
+   "metadata": {},
+   "source": [
+    "## Define the LDM Sampler, which contains functions that will perform the inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8685da6e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ldm_sampler = LDMSampler(\n",
+    "    args.body_region,\n",
+    "    args.anatomy_list,\n",
+    "    args.all_mask_files_json,\n",
+    "    args.all_anatomy_size_condtions_json,\n",
+    "    args.all_mask_files_base_dir,\n",
+    "    args.label_dict_json,\n",
+    "    args.label_dict_remap_json,\n",
+    "    autoencoder,\n",
+    "    diffusion_unet,\n",
+    "    controlnet,\n",
+    "    noise_scheduler,\n",
+    "    scale_factor,\n",
+    "    mask_generation_autoencoder,\n",
+    "    mask_generation_diffusion_unet,\n",
+    "    mask_generation_scale_factor,\n",
+    "    mask_generation_noise_scheduler,\n",
+    "    device,\n",
+    "    latent_shape,\n",
+    "    args.mask_generation_latent_shape,\n",
+    "    args.output_size,\n",
+    "    args.output_dir,\n",
+    "    args.controllable_anatomy_size,\n",
+    "    image_output_ext=args.image_output_ext,\n",
+    "    label_output_ext=args.label_output_ext,\n",
+    "    spacing=args.spacing,\n",
+    "    num_inference_steps=args.num_inference_steps,\n",
+    "    mask_generation_num_inference_steps=args.mask_generation_num_inference_steps,\n",
+    "    random_seed=args.random_seed,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ff93fa3-da0c-46ff-8c77-ab53f39f26f5",
+   "metadata": {},
+   "source": [
+    "## Perform the inference. \n",
+    "It will take around 80s to generate a [256,256,256] mask, and another 105s to generate the corresponding [256,256,256] image on one 80G A100. The time cost per image is roughly linear to the output size. i.e., if we want to generate a [512,512,512] image/mask pair, it will take around 80s to generate a [256,256,256] mask and then resample it to a [512,512,512] mask. The time for [512,512,512] image generatation will be 8 times longer, around 8x105s~14min. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "271f91bf-1c55-46e2-ae56-8677cd8eb81f",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "print(f\"The generated image/mask pairs will be saved in {args.output_dir}.\")\n",
```

Comment:
Could you please help add 1 visualization cell to show the generated image & mask in this notebook? You can refer to @yiheng-wang-nv 's previous example.

Thanks.

Reply:
I was asking whether @guopengf can move the plotting util funcs to somewhere like scripts/utils_plot.py so I can reuse those plotting funcs

Reply:
Done. Thanks